[14.0][MIG] base_delivery_carrier_files

StefanRijnhart opened the pull request.

nbessi committed:
[MRG] 6.1
(lp:c2c-addons/6.1  rev 28.1.5)

Commit:
[FIX] license headers
(lp:c2c-addons/6.1  rev 48)

Commit:
[MRG] from upstream
(lp:c2c-addons/6.1  rev 45.1.4)

Commit:
[FIX] handle no str or unicode values
(lp:c2c-addons/6.1  rev 45.1.5)

Commit:
[FIX] handle no str or unicode values
(lp:c2c-addons/6.1  rev 50)

nbessi committed:
[MRG] from trunk
(lp:c2c-addons/6.1  rev 28.1.7)

Commit:
[MIGR] set installable to False on all the modules, the flag has to be reactivated when the modules are migrated

Commit:
Launchpad automatic translations update.

rdeheele committed:
[UPD] base_delivery_carrier_files addon: port in v7

rdeheele committed:
[FIX] small changes after functional test to generate a correctly filled file

rdeheele committed:
[UPD] keeps country_code term: more logical and as 6.1

rdeheele committed:
[UPD] changes needed to fit with stock.picking/stock.picking.out changes in v7

rdeheele committed:
[UPD] needed changes(bis) to fit with stock.picking/stock.picking.out changes in v7

Commit:
Launchpad automatic translations update.

rdeheele committed:
[UPD] clean according flake8

rdeheele committed:
[FIX] moves copy on stock.picking.out object, adds a test to check duplication

rdeheele committed:
[FIX] fix #1298300, need to add field state in tree description of picking_ids field

Commit:
Launchpad automatic translations update.

rdeheele committed:
[UPD] pass picking_id in context in order to associate precisely attachment and picking

Commit:
Launchpad automatic translations update.

albertofl committed:
[ADD] Add es translation

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

albertofl committed:
Update translations

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

Commit:
Launchpad automatic translations update.

yvaucher committed:
set installable to false

yvaucher committed:
move modules in __unported__

sbidoul committed:
[MOV] move addons out of __unported__ (they remain not installable)

kkarolis committed:
Port base_delivery_carrier_files_migration

oca-transbot committed:
OCA Transbot updated translations from Transifex

Commit:
[REF] make selection function overwritable

pedrobaeza committed:
[MIG] Rename manifest files

sbidoul committed:
remove obsolete .pot files [ci skip]

oca-travis committed:
[UPD] Update base_delivery_carrier_files.pot

Commit:
[FIX] linting and style issues base_delivery_carrier_files

Commit:
[MIG] base_delivery_carrier_files: tests to Python tests

OCA-git-bot committed:
[ADD] icon.png

mFlayyan committed:
[MIG] base_delivery_carrier_files: Migration to 10.0

Commit:
[MIG] base_delivery_carrier_files: Migration to V12.0

oca-travis committed:
[UPD] Update base_delivery_carrier_files.pot

OCA-git-bot committed:
[UPD] README.rst

StefanRijnhart committed:
[IMP] base_delivery_carrier_files: black, isort, prettier

StefanRijnhart committed:
[MIG] base_delivery_carrier_files: Migration to 14.0

hailangvn commented on base_delivery_carrier_files/models/stock_move.py:
```diff
@@ -0,0 +1,19 @@
+    def write(self, values):
+        write_result = super(StockMove, self).write(values)
+        if values.get("state") == "done":
+            done_pickings = self.mapped("picking_id").filtered(
+                lambda sp: sp.state == "done"
+            )
```
May I know the benefit of using filtered+lambda over search+domain? Thanks

hailangvn replied:
Example: https://github.com/odoo/odoo/blob/c08082d89782b7675e8e649cba15ca9e03e40077/addons/purchase_stock/tests/test_replenish_wizard.py#L51

StefanRijnhart replied:
Good question! Because the processes of pickings and their stock moves are very much intertwined, it is likely that at this point we will already have the picking states in our cache so then we avoid the search as well as a potential read. Also, when setting the state of more than one stock move, it is unlikely that there are a large number of pickings involved. I would not prefer to use `filtered` if the result set was potentially large (but it also depends on whether the column used in the domain is indexed, which happens to be the case here).

hailangvn reviewed (approved):
Code review: LGTM. Thank you.

StefanRijnhart committed:
[IMP] base_delivery_carrier_files: pass pickings to _write_file

_write_file was designed to be inheritable to implement other file storages,
but without access to the pickings it is not possible to implement attachment
or binary fields storage.

silverf0x0 commented:
Hi there. What is missing here. Any help needed?

StefanRijnhart commented:
@silverf0x0 this is a standard migration PR (and a tiny API change when storing the output, in the last commit). It's waiting for a second review.

silverf0x0 commented:
@StefanRijnhart Understood. Thank you.

StefanRijnhart commented:
@silverf0x0 by the way, that review could come from you if you have the time!

silverf0x0 commented:
@StefanRijnhart Nice to ear that, because in this very moment I'm testing your work. Any specifc procedure is needed for the review, or I just need to be sure that everything is working?

StefanRijnhart commented:
@silverf0x0 Great, thanks! If you successfully tested the module, please mention that in your approval comment. No specific procedure applies.

silverf0x0 commented:
@StefanRijnhart Excellent, so I'll finish all the tests and I'll approve the review. Thanks to you!

silverf0x0 commented:
@StefanRijnhart I've completed to test everything and for me work/look nice. Maybe the only thing we could add, as improvement, is a UserError (or ValidationError) withing models/stock_picking.py, within generate_carrier_files function:

- when a carrier is not present in an outgoing picking it would not generate a file, as per standard. In this case would be useful to let the user know why a file is not being generated (but in this case the raise could block the generation of other files, which is a little bit annoying). Maybe we can just leave a to-do in the comment?

Thank you in advance

silverf0x0 reviewed (approved):
Tested locally several times, without issues.
Code has been reviewed and is fine.

gfcapalbo commented on base_delivery_carrier_files/__manifest__.py:
```diff
@@ -0,0 +1,24 @@
+    "author": "Camptocamp,Odoo Community Association (OCA)",
+    "license": "AGPL-3",
+    "website": "https://github.com/OCA/delivery-carrier",
+    "depends": ["base", "stock", "delivery"],
```
in v14 delivery implies stock
delivery -> sale_stock ->  stock_account -> stock

gfcapalbo commented on base_delivery_carrier_files/models/stock_picking.py:
```diff
@@ -0,0 +1,60 @@
+    _inherit = "stock.picking"
+
+    def generate_carrier_files(self, auto=True, recreate=False):
+        """
```
fails silently when a picking with  no carrier or no carrier.carrier_file_id  is selected.
Perhaps some feedback (like a warning icon), or  just filtering out non-eligible stock.picking from wizard could be a usability improvement.

gfcapalbo reviewed (approved):
Tested locally, generated the CSV, all looks good , just some observations on usability, those can be discarded because the module will be used by other modules that can add usability niceties themselves.
LGTM.

dreispt commented:
/ocabot merge nobump
